main/SQLHelper.cpp

```diff
@@ -3553,12 +3556,34 @@ uint64_t CSQLHelper::UpdateValueInt(const int HardwareID, const char* ID, const
 					(sValue == old_sValue)
 					);
 			}
+			else if (stype == STYPE_BlindsPercentageInverted)
+			{
+				int level = 0;
+				std::stringstream ssLevel(sValue);
+				ssLevel >> level;
+
+				if (nValue == gswitch_sOn)
+					level = 0;
+				else if (nValue == gswitch_sOff)
+					level = 100;
+				else
+				{
+					if (level == 100)
+						nValue2 = gswitch_sOff;
+					else
+						level = 100 - level; // received an inverted level, inverting back to original
+				}
+				ssLevel.str(std::string());
+				ssLevel.clear();
+				ssLevel << level;
+				ssLevel >> sValue2;
+			}
 
 			result = safe_query(
 				"UPDATE DeviceStatus SET SignalLevel=%d, BatteryLevel=%d, nValue=%d, sValue='%q', LastUpdate='%04d-%02d-%02d %02d:%02d:%02d' "
 				"WHERE (ID = %" PRIu64 ")",
 				signallevel, batterylevel,
-				nValue, sValue,
+				nValue2, sValue2.c_str(),
```

Comment:
The ZWaveBase calls pushes directly to the RX queue, so it's not using SwitchLightInt.. Which is the whole reason for me to use the ZWaveBase at first, otherwise all could be done in this function. I did comment before, but these seem to be lost because of my error :( but using the SwitchLighInt function in ZwaveBase would introduce even more overhead as you stated before with the SQL query, so this is IMO the best solution...